posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -0,0 +1,654 @@
+            tg.create_task(
+                update_saved_query_status(label, DataWarehouseSavedQuery.Status.FAILED, run_at, inputs.team_id)
+            )
+
+
+async def update_saved_query_status(
+    label: str, status: DataWarehouseSavedQuery.Status, run_at: dt.datetime, team_id: int
+):
+    filter_params: dict[str, int | str | uuid.UUID] = {"team_id": team_id}
+
+    try:
+        model_id = uuid.UUID(label)
+        filter_params["id"] = model_id
+    except ValueError:
+        filter_params["name"] = label
+
+    saved_query = await database_sync_to_async(DataWarehouseSavedQuery.objects.filter(**filter_params).get)()
+    saved_query.last_run_at = run_at
+    saved_query.status = status
+
+    await database_sync_to_async(saved_query.save)()
+
+
+@dataclasses.dataclass
+class RunWorkflowInputs:
+    """Inputs to `RunWorkflow`.
+
+    Attributes:
+        team_id: The ID of the team we are running this for.
+        select: A list of model selectors to define the models to run.
+    """
+
+    team_id: int
+    select: list[str] = dataclasses.field(default_factory=list)
+
+
+@temporalio.workflow.defn(name="data-modeling-run")
+class RunWorkflow(PostHogWorkflow):
+    """A Temporal Workflow to run PostHog data models.
+
+    A model is defined by a label, a saved query that dictates how to select the data that
+    makes up the model, and the path or paths to the model through all of its ancestors.
+    """
+
+    @temporalio.workflow.run
+    async def run(self, inputs: RunWorkflowInputs) -> Results:
+        build_dag_inputs = BuildDagActivityInputs(team_id=inputs.team_id, select=inputs.select)
+        dag = await temporalio.workflow.execute_activity(
+            build_dag_activity,
+            build_dag_inputs,
+            start_to_close_timeout=dt.timedelta(minutes=5),
+            retry_policy=temporalio.common.RetryPolicy(
+                initial_interval=dt.timedelta(seconds=10),
+                maximum_interval=dt.timedelta(seconds=60),
+                maximum_attempts=0,
+            ),
+        )
+
+        run_at = dt.datetime.now(dt.UTC).isoformat()
+
+        start_run_activity_inputs = StartRunActivityInputs(dag=dag, run_at=run_at, team_id=inputs.team_id)
+        await temporalio.workflow.execute_activity(
+            start_run_activity,
+            start_run_activity_inputs,
+            start_to_close_timeout=dt.timedelta(minutes=5),
+            retry_policy=temporalio.common.RetryPolicy(
+                initial_interval=dt.timedelta(seconds=10),
+                maximum_interval=dt.timedelta(seconds=60),
+                maximum_attempts=0,
+            ),
+        )
+
+        run_model_activity_inputs = RunDagActivityInputs(team_id=inputs.team_id, dag=dag)
+        results = await temporalio.workflow.execute_activity(
+            run_dag_activity,
+            run_model_activity_inputs,
+            start_to_close_timeout=dt.timedelta(minutes=5),
```

Comment:
Is 5 mins long enough for this? Some queries may take a long time to calculate, and there could be a large chain of nodes to go through first?

Reply:
But maybe it should be an input to the workflow? I honestly have no notion of what timeout should be set to. Unfortunately temporal requires a timeout be set, so I can't leave it empty.